lib/home/movie_service.dart

```diff
@@ -1,34 +1,46 @@
 import 'package:dio/dio.dart';
-import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:flutter_command/flutter_command.dart';
+import 'package:get_it/get_it.dart';
 import 'package:movieapp/environment_config.dart';
 import 'package:movieapp/home/movies_exception.dart';
 
 import 'movie.dart';
 
-final movieServiceProvider = Provider<MovieService>((ref) {
-  final config = ref.watch(environmentConfigProvider);
-
-  return MovieService(config, Dio());
-});
 
 class MovieService {
-  MovieService(this._environmentConfig, this._dio);
+  MovieService([this._environmentConfig, this._dio]) {
+    _environmentConfig ??= GetIt.I<EnvironmentConfig>();
+    _dio ??= Dio();
+  }
 
-  final EnvironmentConfig _environmentConfig;
-  final Dio _dio;
+  EnvironmentConfig _environmentConfig;
+  Dio _dio;
 
   Future<List<Movie>> getMovies() async {
+    await Future.delayed(Duration(milliseconds: 500));
```

Comment:
Is this needed?

Reply:
Without it you won't see any progress indicator but I will add a comment

Reply:
I think that if they really want to see it they can add it themselves. I have a feeling people will try it out and just notice that it "loads slower"

Reply:
Don't think so without they will wonder why doesn't the command work as expected.

Reply:
Have a hard time seeing that but the implementation is good either way :)